README.md

```diff
@@ -1,35 +1,259 @@
+
+```yml
+sqitch_migrations_system_user: nomad
+```
+
+### sqitch_migrations_home
+
+This is a variable defined for convenience as it is used very often.  It defines the home directory of the `sqitch_migrations_system_user`.
+
+```yml
+sqitch_migrations_home: "/home/{{ sqitch_migrations_system_user }}"
+```
+
+### sqitch_migrations_system_dependencies
+
+This variable defines a list packages that should be installed as part of setting up the role.
+
+```yml
+# system-wide dependencies
+sqitch_migrations_system_dependencies:
+  - git
+```
+
+### sqitch_migrations_log_path
+
+This is the directory in which logs will be stored.
+
+```yml
+sqitch_migrations_log_path: "/var/log/sqitch_migrations"
+```
+
+### sqitch_migrations_repositories
+
+A list of git repositories to clone, the database migrations we want to run are located here.
+
+```yml
+sqitch_migrations_repositories:
+  - url: git@github.com:onaio/data-solutions.git
+    destination: "{{ sqitch_migrations_home }}/data-solutions"
+    version: master
+```
+
+### sqitch_migrations_db_connections
+
+An object containing database connections to use when running the role.
+
+```yml
+sqitch_migrations_db_connections:
+  db_admin_user:  # used to identify this particular connection
+    user: db_username
+    password: hunter2
+    host: localhost
+    port: 5432
+    database: dbname
+  db_other_user:  # used to identify this particular connection
+    user: othe_ruser
+    password: hunter2
+    host: localhost
+    port: 5432
+    database: dbname2
+```
+
+### sqitch_migrations_sql_for_setup
+
+An object describing files containing SQL queries that should be run BEFORE database migrations.
+
+```yml
+sqitch_migrations_sql_for_setup:
+  db_admin_user:  # used to identify the db connection to use
+    - absolute-path-to-file-containing-sql
+    - /tmp/some-file.sql
+```
+
+### sqitch_migrations_sqitch_plans
+
+An object containing the actual sqitch database migrations to be run.
+
+```yml
+sqitch_migrations_sqitch_plans:
+  db_admin_user:  # used to identify the db connection to use
+    - directory: "{{ sqitch_migrations_home }}/data-solutions/examples/sqitch_test"
+    - directory: "{{ sqitch_migrations_home }}/data-solutions/1-utils"
+      registry: sqitch_test
+      extra_args: --verify --set schema=test
+      engine: pg
+```
+
+As you can tell, it is an object/dictionary where each entry contains a list of other objects.  The structure of these inner objects is as follows:
+
+- **directory**: absolute path to a directory that contains a sqitch plan
+- **registry**: (optional) the sqitch registry to use, default is "sqitch"
+- **extra_args**: (optional) extra args for sqitch deploy command, default is "--verify"
+- **engine**: (optional) database engine to use, default is pg
+
+### sqitch_migrations_sql_jobs
+
+An object containing SQL queries that should be executed periodically.
+
+```yml
+sqitch_migrations_sql_jobs:
```

Comment:
yay!

IMO we should move to as db-centered an approach as we can, so we can define mat views alongside their refresh intervals (something like a system.cron table or pg_cron) vs requiring redeploy, but we can then just use this to run a SP or something to check dynamically what needs to happen in the next time slice.

Reply:
Yes I agree.  We should push as much as possible to the DB itself.  However I think when we are using Amazon's RDS our hands are a little tied because they don't allow pg_cron.  Anyway, I think we are generally on the same page and should watch out that this is not abused.